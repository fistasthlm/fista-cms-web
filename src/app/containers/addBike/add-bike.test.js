import React from 'react';
import { shallow } from 'enzyme';
import { Map } from 'immutable';
import AddBike from './add-bike';

const setup = propOverrides => {
    const props = Object.assign({
        authenticated: true,
        user: Map(),
        addBike: jest.fn(),
    }, propOverrides);

    return shallow(<AddBike {...props} />);
};

describe('Render', () => {
    it('should render component', () => {
        const wrapper = setup();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render bike form', () => {
        const wrapper = setup({
            user: Map({
                instagram: 'peteholmberg',
            })
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render redirect if not authenticated', () => {
        const wrapper = setup({
            authenticated: false,
        });

        expect(wrapper).toMatchSnapshot();
    });
});


describe('State', () => {
    it('should add bike', () => {
        const wrapper = setup();

        wrapper.instance().saveBike();

        expect(wrapper.instance().props.addBike).toHaveBeenCalled();
    });
});
