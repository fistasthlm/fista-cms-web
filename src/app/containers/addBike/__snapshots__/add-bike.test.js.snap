// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render bike form 1`] = `
<div
  className="add-bike"
>
  <div>
    <h1>
      New biek day
    </h1>
    <p
      className="add-bike__preamble"
    >
      fira mit ein vätska! En titel och minst en bild är kravet.
    </p>
    <BikeForm
      onSubmit={[Function]}
      user={
        Immutable.Map {
          "instagram": "peteholmberg",
        }
      }
    />
  </div>
</div>
`;

exports[`Render should render component 1`] = `
<div
  className="add-bike"
/>
`;

exports[`Render should render redirect if not authenticated 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/login",
      "state": Object {
        "from": undefined,
      },
    }
  }
/>
`;
